pkg/api/v1/externalreference.go

```diff
@@ -3,5 +3,6 @@ package v1
 type ExternalProjectReference struct {
 	// ID is the Atlas project ID
 	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern:=`^([a-f0-9]{24})$`
```

Comment:
You cannot retrospectively tighten validation. This potentially can break existing users.

Reply:
we can do this but only once CRD ratcheting is 100% supported by the minimal Kubernetes version, cc @roothorp

Reply:
This is required by Atlas, so it has to be in this format already. But I can remove it, no problem

Reply:
Removed